src/module/skb/bpf.rs

```diff
@@ -318,24 +296,40 @@ pub(super) fn unmarshal_packet(event: &mut SkbEvent, raw_section: &BpfRawSection
 
     match eth.get_ethertype() {
         EtherTypes::Arp => {
-            event.arp = unmarshal_arp(
-                &ArpPacket::new(eth.payload())
-                    .ok_or_else(|| anyhow!("Could not parse ARP packet"))?,
-            )?;
+            if let Some(eth) = ArpPacket::new(eth.payload()) {
+                event.arp = unmarshal_arp(&eth)?;
+            };
         }
         EtherTypes::Ipv4 => {
-            event.ip = Some(unmarshal_ipv4(
-                &Ipv4Packet::new(eth.payload())
-                    .ok_or_else(|| anyhow!("Could not parse IPv4 packet"))?,
-            )?)
+            if let Some(ip) = Ipv4Packet::new(eth.payload()) {
+                event.ip = Some(unmarshal_ipv4(&ip)?);
+                unmarshal_l3(event, ip.get_next_level_protocol(), ip.payload())?;
+            };
         }
         EtherTypes::Ipv6 => {
-            event.ip = Some(unmarshal_ipv6(
-                &Ipv6Packet::new(eth.payload())
-                    .ok_or_else(|| anyhow!("Could not parse IPv6 packet"))?,
-            )?)
+            if let Some(ip) = Ipv6Packet::new(eth.payload()) {
+                event.ip = Some(unmarshal_ipv6(&ip)?);
+                unmarshal_l3(event, ip.get_next_header(), ip.payload())?;
+            };
+        }
+        _ => (),
+    }
+
+    Ok(())
+}
+
+fn unmarshal_l3(
```

Comment:
Funny name for a function that unmarshals the L4 header :-)

Reply:
:rofl: